File: Merge K sorted linked lists.java
class Solution {
    // Function to merge K sorted linked list.
    Node mergeKLists(List<Node> arr) {
        // Add your code here.
        PriorityQueue<Node>pq=new PriorityQueue<>((a,b) -> a.data-b.data);
        for(int i=0;i<arr.size();i++){
            pq.add(arr.get(i));
        }
        Node head1=new Node(-1);
        Node dummy=head1;
        while(!pq.isEmpty()){
            Node minNode=pq.poll();
            head1.next=minNode;
            head1=head1.next;
            if(minNode.next!=null){
                pq.add(minNode.next);
            }
        }
        return dummy.next;
    }
}

File: Merge K sorted linked lists.cpp
class Solution {
  public:
    Node* mergeKLists(vector<Node*>& arr) {
        // code here
        bool check=true;
        Node* st=new Node(-1);
        Node* head=st;
        while(check){
            int minNum=INT_MAX;
            int mini=-1;
            check=0;
            for(int i=0;i<arr.size();i++){
                if(arr[i] && arr[i]->data<minNum){
                    minNum=arr[i]->data;
                    mini=i;
                    check=1;
                }
                
            }
            if(mini!=-1){
                Node* temp=new Node(minNum);
                st->next=temp;
                st=temp;
                arr[mini]=arr[mini]->next;
            }
        }
        return head->next;
    }
};
